src/v/cluster/persisted_stm.h

```diff
@@ -103,11 +115,14 @@ class file_backed_stm_snapshot {
  * and uses it as a base for replaying the commands.
  */
 
+template<supported_stm_backend T = file_backed_stm_snapshot>
```

Comment:
nit: i may be a little picky but it is hard to tell from the first read that `file_backed_stm_snapshot` is `supported_stm_backend`, maybe we can call call it `file_based_stm_backend` or rename the concept to `supported_stm_snapshot_backend`